documentation/developer-guide/modules/tooling-guide/pages/java-aspect-tooling.adoc

```diff
@@ -314,6 +314,26 @@ To use the generated static meta classes, you need the following additional depe
 
 include::sds-developer-guide:ROOT:partial$sds-aspect-static-meta-model-java-artifact.adoc[]
 
+==== Providing Custom Macros for Code Generation
+
+It is possible to change predefined sections of the generated classes by providing custom https://velocity.apache.org/[Velocity] template http://people.apache.org/~henning/velocity/html/ch07.html[macros].
```

Comment:
I think I'd rather link to https://velocity.apache.org/engine/1.7/user-guide.html instead.

Reply:
Would you replace the link to the Velocity as such or to the macro documentation?

Reply:
No, I'd remove the second link (looking at the URL, I'm not sure if that is an official documentation). E.g., "It is possible to change predefined sections of the generated classes by providing custom https://velocity.apache.org/[Velocity] templates; see the https://velocity.apache.org/engine/1.7/user-guide.html[Velocity User Guide] for more information".